File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.js
function solution(participant, completion) {
    const map = new Map();
    
    // 참가자 수 세기
    for (const name of participant) {
        map.set(name, (map.get(name) || 0) + 1)
    }
    
    // 완주자 수 세기
    for (const name of completion) {
        map.set(name, map.get(name) - 1)
    }
    
    // 3. 카운트가 0이 아닌 사람 찾기
    for (const [name, count] of map) {
        if (count > 0) return name;
    }
    
}

File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.py
from collections import defaultdict
def solution(participant, completion):
    answer = ''
    people = defaultdict(int)
    
    for name in participant:
        people[name] += 1
    
    for name in completion:
        people[name] -= 1
    
    # print(people)
    # print(list(people.items()))
    # print(dict(people))
    
    for k, v in people.items():
        if v >= 1:
            answer = k
    
    return answer